### PR TITLE
Add collapse sample

### PR DESCRIPTION
**Pull request by araithel**

### CODE REVIEW

File: src/main/java/de/picturesafe/search/samples/collapse/Collapse.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            singleIndexElasticsearchService.deleteIndexWithAlias();
+        }
+    }
+
+    private void findTweetsSortByLikes(Expression expression) {
+        final SearchParameter searchParameter = SearchParameter.builder().sortOptions(SortOption.desc("likes")).build();
+        search(expression, searchParameter);
+    }
+
+    private void findMostLikedTweetOfEachUserSortByLikes(Expression expression) {
```
Comment (jfrommann) [inline src/main/java/de/picturesafe/search/samples/collapse/Collapse.java:83]: 1. There is only one user, so this example may not be the best.
2. Doesn't "most liked" imply that it's sort by likes? Why doubling it in the method name?

Comment (araithel) [reply src/main/java/de/picturesafe/search/samples/collapse/Collapse.java:83]: There are two users: "Jane Doe" and "John Doe" ;-)

The users are sorted by most likes AND the document with the highest like is returned (for each user).
For example, if the users were sorted by name (user), the method would be 'findMostLikedTweetOfEachUserSortByUser'.

So it's okay for me.

Comment (jfrommann) [reply src/main/java/de/picturesafe/search/samples/collapse/Collapse.java:83]: Hmm, the names look very similar, that's why i didn't see the difference. Wouldn't it be better to have obviously different names?

Comment (araithel) [reply src/main/java/de/picturesafe/search/samples/collapse/Collapse.java:83]: Do you have a good idea? How about:

findTweetsSortByLikes
collapseMostLikedTweetForEachUser
expandMostLikedTweetsOfEachUser

Comment (jfrommann) [reply src/main/java/de/picturesafe/search/samples/collapse/Collapse.java:83]: OK, i should have made clearer that i mean the user names. Sorry!

What about "John Doe" and "Jeanne d’Arc"

Comment (araithel) [reply src/main/java/de/picturesafe/search/samples/collapse/Collapse.java:83]: ok, done!

**Pull request merged**